### ISSUE TITLE
Add a task template to Create Class to reboot machine by name

### ISSUE THREAD

**Issue by uwardlaw** [dev4vater/vater#27]
- [x] Blocked by #95 
- [ ] Add `rebootVM.yml` to `rous/ansible`
- [x] Add `Reboot VM` task template to `setupNewClass.py`

**Comment by marissaeinhorn** [comment]
[Reference](https://www.robvit.com/ansible-tower-awx/ansible-vmware-playbook-examples/)  Want to stop VM, wait for 2 seconds and then power on VM instead of using built in restart option.

Stop a VM:
```
- name: VMWare tasks
  hosts: localhost
  connection: local
  gather_facts: no
  vars:
    vcenter_ip: 'VCENTERIP'
    vcenter_user: 'VCENTERUSER'
    vcenter_password: 'VCENTERPASS'
    vmname: 'VMNAME'
  tasks:
    - name: Wait for the virtual machine to shutdown
      vmware_guest_powerstate:
        hostname: "{{ vcenter_ip }}"
        username: "{{ vcenter_user }}"
        password: "{{ vcenter_password }}"
        name: "{{ vmname }}"
        validate_certs: False
        state: shutdown-guest
        state_change_timeout: 900
      register: wait
      ignore_errors: yes

    - name: ForcePoweroff 
      vmware_guest_powerstate:
        hostname: "{{ vcenter_ip }}"
        username: "{{ vcenter_user }}"
        password: "{{ vcenter_password }}"
        name: "{{ vmname }}"
        validate_certs: False
        state: powered-off
      register: poweroff
      when: wait.msg is search('Timeout')
```

Start a VM:
```
- name: VMWare tasks
  hosts: localhost
  connection: local
  gather_facts: no
  vars:
    vcenter_ip: 'VCENTERIP'
    vcenter_user: 'VCENTERUSER'
    vcenter_password: 'VCENTERPASS'
    vmname: "VMNAME"
  tasks:
    - name: ForcePowerOn
      vmware_guest_powerstate:
        hostname: "{{ vcenter_ip }}"
        username: "{{ vcenter_user }}"
        password: "{{ vcenter_password }}"
        name: "{{ vmname }}"
        validate_certs: False
        state: powered-on
```

**Comment by marissaeinhorn** [comment]
- [ ] Update task template to contain variables for `vmName`, `enclave`, and `studentNumber`

**Comment by marissaeinhorn** [comment]
Error when creating tasks 
```

TASK [Setup New Class Project] *************************************************
1:05:32 PM
fatal: [localhost]: FAILED! => changed=true
1:05:32 PM
cmd:
1:05:32 PM
- python3
1:05:32 PM
- ../bin/setupNewClass.py
1:05:32 PM
- ve72dhxquo9sjcvmox5hpkjfefomb7q4iwbf5xu9r2s=
1:05:32 PM
- '8'
1:05:32 PM
- 192.168.100.20
1:05:32 PM
- http://192.168.100.20:3000/333TRS/rous
1:05:32 PM
- tasks
1:05:32 PM
- /home/control/rous/vms/
1:05:32 PM
- /home/control/rous/terraform/
1:05:32 PM
- U22002
1:05:32 PM
- '2'
1:05:32 PM
delta: '0:00:00.403244'
1:05:32 PM
end: '2022-02-16 19:05:32.070790'
1:05:32 PM
msg: non-zero return code
1:05:32 PM
rc: 1
1:05:32 PM
start: '2022-02-16 19:05:31.667546'
1:05:32 PM
stderr: ''
1:05:32 PM
stderr_lines: <omitted>
1:05:32 PM
stdout: |-
1:05:32 PM
[{'id': 13,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'NoneKey',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'None'}]
1:05:32 PM
Found ID: 13
1:05:32 PM
[{'id': 14,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'control',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'ssh'},
1:05:32 PM
{'id': 13,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'NoneKey',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'None'}]
1:05:32 PM
Found ID: 14
1:05:32 PM
[{'id': 14,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'control',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'ssh'},
1:05:32 PM
{'id': 13,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'NoneKey',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'None'},
1:05:32 PM
{'id': 15,
1:05:32 PM
'login_password': {'login': '', 'password': ''},
1:05:32 PM
'name': 'NoneKeyLP',
1:05:32 PM
'override_secret': False,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh': {'login': '', 'passphrase': '', 'private_key': ''},
1:05:32 PM
'type': 'login_password'}]
1:05:32 PM
Found ID: 15
1:05:32 PM
[{'git_url': 'http://192.168.100.20:3000/333TRS/rous',
1:05:32 PM
'id': 5,
1:05:32 PM
'name': 'Playbooks',
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh_key_id': 13}]
1:05:32 PM
Found ID: 5
1:05:32 PM
[{'become_key_id': None,
1:05:32 PM
'id': 7,
1:05:32 PM
'inventory': '[control]\n192.168.100.20',
1:05:32 PM
'name': 'control',
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False,
1:05:32 PM
'ssh_key_id': 14,
1:05:32 PM
'type': 'static'}]
1:05:32 PM
Found ID: 7
1:05:32 PM
[{'id': 5,
1:05:32 PM
'json': '{"classNumber": "U22002","classSize": 2,"ansiblePathInRepository": '
1:05:32 PM
'"tasks","vmPathInRepository": '
1:05:32 PM
'"/home/control/rous/vms/","terraformPathInRepositoryOnControl": '
1:05:32 PM
'"/home/control/rous/terraform/"}',
1:05:32 PM
'name': 'Env',
1:05:32 PM
'password': None,
1:05:32 PM
'project_id': 8,
1:05:32 PM
'removed': False}]
1:05:32 PM
Found ID: 5
1:05:32 PM
Response 400, Bad request
1:05:32 PM
stdout_lines: <omitted>
1:05:32 PM
1:05:32 PM
PLAY RECAP *********************************************************************
1:05:32 PM
localhost : ok=5 changed=0 unreachable=0 failed=1 skipped=4 rescued=0 ignored=0
1:05:32 PM
1:05:32 PM
Running playbook failed: exit status 2
```

**Comment by uwardlaw** [comment]
Looks like a bad request on the create Env. Is there a branch with this
code?

Issue is likely in `rous/bin/createNewClass.py` around line 182-194

**Comment by marissaeinhorn** [comment]
Tasks create successfully with variables `student`, `enclave` and `vmName` 

`restartVm.yml` playbook executes with the following error: 

TASK [Wait for the virtual machine to shutdown] ********************************
8:48:08 AM
fatal: [localhost]: FAILED! => changed=false
8:48:08 AM
details: Please see documentation of the vmware_guest module for folder parameter.
8:48:08 AM
msg: Multiple virtual machines with same name [windows_fundamentals_router] found, Folder value is a required parameter to find uniqueness of the virtual machine
8:48:08 AM
...ignoring

**Comment by marissaeinhorn** [comment]
```
'Unsupported parameters for (vmware_guest_powerstate) module: datacenter Supported parameters include: folder, force, hostname, moid, name, name_match, password, port, proxy_host, proxy_port, schedule_task_description, schedule_task_enabled, schedule_task_name, scheduled_at, state, state_change_timeout, use_instance_uuid, username, uuid, validate_certs
```

**Comment by marissaeinhorn** [comment]
Task template is working, need to include logic to add leading zero to student number for consistency

**Comment by marissaeinhorn** [comment]
Formatting student numbers [ref](https://stackoverflow.com/questions/18050911/how-to-format-a-variable-in-ansible-value)  Successful test with limited network, updated vm naming convention currently commented out in iss27 branch.

**Comment by marissaeinhorn** [comment]
Merged into iss180 branch.  Tested and is working
